qiskit/algorithms/eigen_solvers/vqd.py

```diff
@@ -0,0 +1,840 @@
+from ..exceptions import AlgorithmError
+from ..aux_ops_evaluator import eval_observables
+
+logger = logging.getLogger(__name__)
+
+OBJECTIVE = Callable[[np.ndarray], float]
+GRADIENT = Callable[[np.ndarray], np.ndarray]
+RESULT = Union[scipy.optimize.OptimizeResult, OptimizerResult]
+BOUNDS = List[Tuple[float, float]]
+
+MINIMIZER = Callable[
+    [
+        OBJECTIVE,  # the objective function to minimize (the energy in the case of the VQE)
+        np.ndarray,  # the initial point for the optimization
+        Optional[GRADIENT],  # the gradient of the objective function
+        Optional[BOUNDS],  # parameters bounds for the optimization
+    ],
+    RESULT,  # a result object (either SciPy's or Qiskit's)
+]
+
+# TODO: Add a more detailed explanation of the algorithm in the main docstring
+
+
+class VQD(VariationalAlgorithm, Eigensolver):
+    r"""The Variational Quantum Deflation algorithm.
+
+    `VQD <https://arxiv.org/abs/1805.08138>`__ is a quantum algorithm that uses a
+    variational technique to find
+    the k eigenvalues of the Hamiltonian :math:`H` of a given system.
+
+    An instance of VQD requires defining three algorithmic sub-components:
+    an integer k denoting the number of eigenstates to calculate, a trial
+    state (a.k.a. ansatz)which is a :class:`QuantumCircuit`,
+    and one of the classical :mod:`~qiskit.algorithms.optimizers`.
+    The ansatz is varied, via its set of parameters, by the optimizer,
+    such that it works towards a state, as determined by the parameters
+    applied to the ansatz, that will result in the minimum expectation values
+    being measured of the input operator (Hamiltonian). The algorithm does
+    this by iteratively refining each excited state to be orthogonal to all
+    the previous excited states.
+
+    An optional array of parameter values, via the *initial_point*, may be provided as the
+    starting point for the search of the minimum eigenvalue. This feature is particularly useful
+    such as when there are reasons to believe that the solution point is close to a particular
+    point.
+
+    The length of the *initial_point* list value must match the number of the parameters
+    expected by the ansatz being used. If the *initial_point* is left at the default
+    of ``None``, then VQE will look to the ansatz for a preferred value, based on its
+    given initial state. If the ansatz returns ``None``,
+    then a random point will be generated within the parameter bounds set, as per above.
+    If the ansatz provides ``None`` as the lower bound, then VQE
+    will default it to :math:`-2\pi`; similarly, if the ansatz returns ``None``
+    as the upper bound, the default value will be :math:`2\pi`.
+
+    """
+
+    def __init__(
+        self,
+        ansatz: Optional[QuantumCircuit] = None,
+        k: int = 2,
+        betas: Optional[List[float]] = None,
+        optimizer: Optional[Union[Optimizer, MINIMIZER]] = None,
```

Comment:
Can you take a look at VQE; the MINIMIZER was changed into a protocol and moved into optimizers. So you should just be able to import/use that the same way here and get of the MINIMIZER and its related type definitions above.